tests/storage/snapshot.go

```diff
@@ -626,36 +631,50 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 				_, err = virtClient.VirtualMachineSnapshot(snapshot.Namespace).Create(context.Background(), snapshot, metav1.CreateOptions{})
 				Expect(err).ToNot(HaveOccurred())
 
-				Eventually(func() bool {
+				Eventually(func() *snapshotv1.VirtualMachineSnapshotStatus {
 					snapshot, err = virtClient.VirtualMachineSnapshot(vm.Namespace).Get(context.Background(), snapshot.Name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					return snapshot.Status
+				}, 30*time.Second, 2*time.Second).Should(gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+					"Phase": Equal(snapshotv1.InProgress),
+				})))
+				Eventually(func() string {
 					updatedVMI, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
-					return snapshot.Status != nil &&
-						snapshot.Status.Phase == snapshotv1.InProgress &&
-						updatedVMI.Status.FSFreezeStatus == "frozen"
-				}, 30*time.Second, 2*time.Second).Should(BeTrue())
+					return updatedVMI.Status.FSFreezeStatus
+				}, time.Minute, 2*time.Second).Should(Equal("frozen"))
 
-				contentName := fmt.Sprintf("%s-%s", vmSnapshotContent, snapshot.UID)
-				Eventually(func() bool {
+				Eventually(func() *snapshotv1.VirtualMachineSnapshotStatus {
 					snapshot, err = virtClient.VirtualMachineSnapshot(vm.Namespace).Get(context.Background(), snapshot.Name, metav1.GetOptions{})
 					Expect(err).ToNot(HaveOccurred())
+					return snapshot.Status
+				}, 180*time.Second, 2*time.Second).Should(gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
```

Comment:
What about `matcher.HaveConditionTrue` and others?

Reply:
Like this I can also check the reason

Reply:
I am happy to extend `HaveConditionTrue`

Reply:
yeah thats an idea for another PR :)

Reply:
Fair enough :)